src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/logging/ApplicationEventsEmitter.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerdeletionservice.logging
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Component
+import org.springframework.transaction.event.TransactionalEventListener
+
+@Component
+class ApplicationInsightsEventEmitter(
+  val telemetryClient: TelemetryClient
+) {
+
+  @TransactionalEventListener
+  fun deletionEventListener(deletionEvent: DeletionEvent) {
+    deletionEvent.apply {
+      telemetryClient.trackEvent(
+        event,
+        mutableMapOf("offenderNo" to offenderNumber, "count" to offenderIds?.size.toString()),
```

Comment:
not quite sure how `offenderIds?.size.toString` is being allowed in kotlin - shouldn't it be `offenderIds?.size?.toString` since `offenderIds` can be null so need to chain?